doc/connectivity/bluetooth/api/l2cap.rst

```diff
@@ -65,9 +65,101 @@ listen to, the required security level ``sec_level``, and the callback
+4. Implement the callbacks in :c:type:`bt_l2cap_chan_ops` such as ``recv``, ``connected`` and ``disconnected`` to handle data and connection events.
+
+A minimal skeleton example follows (fill in allocation and error handling as needed):
+
+.. code-block:: c
+
+    /* Minimal L2CAP server example (skeleton) */
+
+    #include <zephyr/bluetooth/bluetooth.h>
+    #include <zephyr/bluetooth/l2cap.h>
+    #include <zephyr/sys/printk.h>
+
+    static int my_chan_recv(struct bt_l2cap_chan *chan, struct net_buf *buf);
+    static void my_chan_connected(struct bt_l2cap_chan *chan);
+    static void my_chan_disconnected(struct bt_l2cap_chan *chan);
+
+    /* Define channel operations */
+    static struct bt_l2cap_chan_ops my_chan_ops = {
+        .recv       = my_chan_recv,
+        .connected  = my_chan_connected,
+        .disconnected = my_chan_disconnected,
+    };
+
+    /* Channel instance (one per connection). Allocate dynamically if you need many connections. */
+    static struct bt_l2cap_le_chan my_chan = {
+        .chan.ops = &my_chan_ops,
+    };
+
+    /* Accept callback: called on an incoming connection request. */
+    static int my_server_accept(struct bt_conn *conn, struct bt_l2cap_chan **chan)
+    {
+        printk("L2CAP accept from %p\n", conn);
+
+        /* Assign our channel instance (or dynamically allocate a new one) */
+        *chan = &my_chan.chan;
+
+        /* return 0 to accept, or negative errno to reject */
+        return 0;
+    }
+
+    static void my_chan_connected(struct bt_l2cap_chan *chan)
+    {
+        printk("L2CAP channel connected\n");
+    }
+
+    static void my_chan_disconnected(struct bt_l2cap_chan *chan)
+    {
+        printk("L2CAP channel disconnected\n");
+    }
+
+    static int my_chan_recv(struct bt_l2cap_chan *chan, struct net_buf *buf)
+    {
+        printk("L2CAP received %u bytes\n", buf->len);
+            /* Process data here; returning 0 tells the stack to free the buffer.
+            * If you need to process asynchronously, return -EINPROGRESS and later
+            * call bt_l2cap_chan_recv_complete(chan, buf) when done (then you must
+            * manage buf ownership yourself).
+            */
+        return 0;
+    }
+
+    static struct bt_l2cap_server my_server = {
+        .psm = BT_L2CAP_PSM_DYNAMIC, /* or a fixed PSM value */
+        .sec_level = BT_SECURITY_LOW,
+        .accept = my_server_accept,
+    };
+
+    /* Call this after Bluetooth is enabled to register the server */
+    static int register_my_server(void)
+    {
+        int err = bt_l2cap_server_register(&my_server);
+        if (err) {
+            printk("Failed to register L2CAP server: %d\n", err);
+            return err;
+        }
+        printk("L2CAP server registered\n");
+        return 0;
+    }
+
+Notes
+~~~~~
+* For multiple concurrent connections allocate channel instances dynamically (k_malloc or pool) instead of reusing a single ``my_chan``.
```

Comment:
You could also allocate channel instances based on the `CONFIG_BT_MAX_CONN` parameter, similar to: https://github.com/zephyrproject-rtos/zephyr/blob/0ab40913c2de53ee504327761b37c367cb544944/doc/connectivity/bluetooth/api/l2cap.rst?plain=1#L17
and 
https://github.com/zephyrproject-rtos/zephyr/blob/0ab40913c2de53ee504327761b37c367cb544944/doc/connectivity/bluetooth/api/l2cap.rst?plain=1#L29-L35